AdminUi/apps/admin_ui/lib/home/identity_details/identity_details.dart

```diff
@@ -0,0 +1,256 @@
+                      _IdentityDetails(
+                        title: 'Client ID',
+                        value: identityDetails.clientId,
+                      ),
+                      _IdentityDetails(
+                        title: 'Public Key',
+                        value: identityDetails.publicKey,
+                      ),
+                      _IdentityDetails(
+                        title: 'Created at',
+                        value: DateFormat('yyyy-MM-dd hh:mm:ss').format(identityDetails.createdAt),
+                      ),
+                      Row(
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          Text(
+                            'Tier: ',
+                            style: Theme.of(context).textTheme.bodyLarge!.copyWith(fontWeight: FontWeight.bold),
+                          ),
+                          DropdownButton<String>(
+                            isDense: true,
+                            value: selectedTier,
+                            onChanged: (String? newValue) {
+                              onTierChanged?.call(newValue);
+                              if (newValue != identityDetails.tierId) {
+                                updateIdentity?.call();
+                              }
+                            },
+                            items: _buildTierDropdownItems(context),
+                          ),
+                        ],
+                      ),
+                    ],
+                  ),
+                ],
+              ),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  List<DropdownMenuItem<String>> _buildTierDropdownItems(BuildContext context) {
+    if (availableTiers.isEmpty) {
+      return [
+        const DropdownMenuItem<String>(
+          value: _noTiersFoundMessage,
+          child: Text(
+            _noTiersFoundMessage,
+            style: TextStyle(fontSize: 10),
+          ),
+        ),
+      ];
+    }
+
+    return availableTiers.where(_isTierManuallyAssignable).map((tier) {
+      final isDisabled = _isTierDisabled(tier);
+      return DropdownMenuItem<String>(
+        value: tier.id,
+        enabled: !isDisabled,
+        child: Text(
+          tier.name,
+          style: TextStyle(
+            color: isDisabled ? Theme.of(context).disabledColor : null,
+            fontSize: 18,
+          ),
+        ),
+      );
+    }).toList();
+  }
+
+  bool _isTierDisabled(TierOverview tier) {
+    final tiersThatCannotBeUnassigned = availableTiers.where((t) => !t.canBeManuallyAssigned);
+    final identityIsInTierThatCannotBeUnassigned = tiersThatCannotBeUnassigned.any((t) => t.id == identityDetails.tierId);
+    return identityIsInTierThatCannotBeUnassigned && tier.id != identityDetails.tierId;
+  }
+
+  bool _isTierManuallyAssignable(TierOverview tier) {
+    return tier.canBeManuallyAssigned || tier.id == identityDetails.tierId;
+  }
```

Comment:
I don't get the second part.

Reply:
The second part is to display for E.g. "Queued for Deletion" Tier otherwise it will not be shown in the Tier dropdown because by default it has the value tier.canBeManuallyAssigned to false. In order to show the QfD tier and also to show other tiers as not selectable (disabled).